### PR TITLE
Test Network Configuration Details

### PR DESCRIPTION
**Pull request by prudmirya**
Test Network Configuration Details

configuration details like speed, driver,transmission mode, link state

Signed-off-by: prudhvi <mprudhvi@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by prudmirya** [comment]
[job.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/582706/job.txt)
log added

File: io/net/Network_config.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        self.log.info("Performing Ethtool and interface checks for interface")
+        self.log.info("-----------------------------------------------------")
+        if speed != eth_speed:
+            self.fail("mis match in speed")
+        hw_addr = netifaces.ifaddresses(self.iface)[netifaces.AF_LINK]
+        hw_addr = hw_addr[0]['addr']
+        if hw_addr != address:
+            self.fail("mismatch in hardware address")
+        cmd = "ifconfig %s" % self.iface
+        mtuval = process.system_output(cmd, shell=True).strip()
```
Comment (PraveenPenguin) [inline io/net/Network_config.py]: can this be 

`mtuval = process.system_output(cmd, shell=True).strip().split('\n')` to eliminate extra variable

File: io/net/Network_config.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+                if mtu != mtu_val:
+                    self.fail("mismatch in mtu")
+            if 'MTU' in val:
+                cmd, mtu_val = val.split('MTU:')
+                mtu_val, cmd = mtu_val.split(' ')
+                self.log.info("through ifconfig mtu value is %s" % mtu_val)
+                if mtu != mtu_val:
+                    self.fail("mismatch in mtu")
+        eth = "ethtool %s | grep 'Link detected:'" % self.iface
+        eth_state = process.system_output(eth, shell=True)
```
Comment (PraveenPenguin) [inline io/net/Network_config.py]: here as well can this be 

`eth_state = process.system_output("ethtool %s | grep 'Link detected:'" % self.iface, shell=True)`

**Comment by prudmirya** [comment]
closed current PR and added new PR #165

**Pull request closed without merging**